src/source/JNIGenerator.scala

```diff
@@ -31,9 +31,9 @@ class JNIGenerator(spec: Spec) extends Generator(spec) {
   val jniBaseLibClassIdentStyle = IdentStyle.prefix("H", IdentStyle.camelUpper)
   val jniBaseLibFileIdentStyle = jniBaseLibClassIdentStyle
 
-  val writeJniCppFile = writeCppFileGeneric(spec.jniOutFolder.get, spec.jniNamespace, spec.jniFileIdentStyle, spec.jniIncludePrefix) _
+  val writeJniCppFile = writeCppFileGeneric(spec.jniOutFolder.get, spec.jniNamespace, spec.jniFileIdentStyle, spec.jniIncludePrefix, spec.cppExt, spec.cppHeaderExt) _
```

Comment:
I don't see where these new args are passed.  Can you clarify what changes you're making to non-Cx generators, and how they're related?